internal/handlers/common/filter.go

```diff
@@ -289,17 +291,34 @@ func filterFieldExpr(fieldValue any, expr *types.Document) (bool, error) {
 
 // filterFieldRegex handles {field: /regex/} filter.
 func filterFieldRegex(fieldValue any, regex types.Regex) (bool, error) {
-	s, ok := fieldValue.(string)
-	if !ok {
-		return false, nil
-	}
+	switch fieldValue := fieldValue.(type) {
+	case string:
+		re, err := regex.Compile()
+		if err != nil {
+			return false, err
+		}
+		return re.MatchString(fieldValue), nil
 
-	re, err := regex.Compile()
-	if err != nil {
-		return false, err
+	case *types.Array:
+		for i := 0; i < fieldValue.Len(); i++ {
+			arrValue := must.NotFail(fieldValue.Get(i)).(any)
+			s, isString := arrValue.(string)
+			if !isString {
+				continue
+			}
+
+			re, err := regex.Compile()
```

Comment:
regex Compile can be done once right at the func beginning

Reply:
fixed